src/datalinker/config.py

```diff
@@ -36,6 +36,15 @@ class Configuration:
     Set with the ``DATALINKER_TOKEN`` environment variable.
     """
 
+    google_credentials: str = os.getenv("GOOGLE_APPLICATION_CREDENTIALS", "")
+    """File to use for credentials when interacting with GCS.
+
+    Set by a kubernetes secret setting this environment variable.
+    The app doesn't use this directly but the GCS library will use it.
+    If this environment variable isn't set, datalinker assumes to use
+    S3 (which has multiple environment variables).
+    """
```

Comment:
The name of this setting confuses me a bit because there are no credentials for Google, since datalinker uses workload identity. So I think this is just used as a boolean variable saying whether to use Google or not. Maybe it should have a different name that reflects that?

Reply:
When I describe the datalinker pod it shows this:

    Environment:
      DATALINKER_CUTOUT_SYNC_URL:      https://data.lsst.cloud/api/cutout/sync
     ..... .... 
      PGPASSFILE:                      /tmp/secrets/postgres-credentials.txt
      S3_ENDPOINT_URL:                 https://storage.googleapis.com
      GOOGLE_APPLICATION_CREDENTIALS:  /tmp/secrets/butler-gcs-idf-creds.json

So I'm using that environment variable which is already set to that file.  I figured if that environment variable is set that means we want to use google.  If we want to use S3 I won't set that environment variable.  I understand that it's not used directly in the code.

Reply:
Oh, huh. I think that's not needed but I'm not actually sure. Yeah, this seems fine for now given that; maybe it's being used under the hood by Butler.

Reply:
So the whole point of this ticket is to work "on-prem" for USDF and the summit.  So I was relying on this environment variable not being set, as of course, the whole point is to not use Google, and to use some S3 setup that is on USDF and the summit.  The ticket didn't say that I need to deploy some on prem S3, and the TAP service as usdf I think uses an on prem S3 service.

But maybe keying off of that environment variable is more dangerous than I thought.  I'm keying off of this  here: 

https://github.com/lsst-sqre/datalinker/blob/14e13d866e50f1ee529aca9f33e55bff488d1761/src/datalinker/handlers/external.py#L271

But maybe I should just make my own environment variable.

Reply:
No, I'm convinced, what you're doing here is fine. I think I confused myself and datalinker is not currently using workload identity so this variable really is in use, just hidden inside the Butler library, so using it this way is probably safe unless USDF is also injecting Google credentials (and I see no reason why they would be doing that).

Correct, the SLAC folks maintain the S3-compatible thing at USDF; we don't have to run it, we're just supposed to use it.